updated receiver docs (#3880)
thanos-io/thanos: docs/components/receive_route.md

We recommend this component to users who can only push into a Thanos due to air-gapped or egress-only environments. Please note the [various pros and cons of pushing metrics](https://docs.google.com/document/d/1H47v7WfyKkSLMrR8_iku6u9VB73WrVzBHb2SB6dL9_g/edit#heading=h.2v27snv0lsur).

For further information on tuning Prometheus Remote Write [see remote write tuning document](https://prometheus.io/docs/practices/remote_write/).

# Example

```bash
thanos receive-route \
    --http-address 0.0.0.0:10909 \
    --receive.replication-factor 1 \
    --receive.hashrings-file ./data/hashring.json \
    --remote-write.address 0.0.0.0:10908 \
```

The example of `remote_write` Prometheus configuration:

```yaml
remote_write:
- url: http://<thanos-receive-route-container-ip>:10908/api/v1/receive
```

where `<thanos-receive-route-container-ip>` is an IP address reachable by Prometheus Server.

The example content of `hashring.json`:

```json
[
    {
        "endpoints": [
            "127.0.0.1:10907",
            "127.0.0.1:11907",
            "127.0.0.1:12907"
        ]
    }
]
```

> With such configuration, the router will listen for remote write requests on `<ip>10908/api/v1/receive` and will forward to the correct receiver in the hashring

Reviewer: ```suggestion
With such configuration, the router will listen for remote write requests on `<ip>:10908/api/v1/receive` and will forward to the correct receiver in the hashring
```
Just a small nit.

                              is specified, there is no client verification on
                              server side. (tls.NoClientCert)
      --remote-write.client-tls-cert=""
                              TLS Certificates to use to identify this client to
                              the server.
      --remote-write.client-tls-key=""
                              TLS Key for the client's certificate.
      --remote-write.client-tls-ca=""
                              TLS CA Certificates to use to verify servers.
      --remote-write.client-server-name=""
                              Server name to verify the hostname on the returned
                              gRPC certificates. See
                              https://tools.ietf.org/html/rfc4366#section-3.1
      --receive.hashrings-file=<path>
                              Path to file that contains the hashring
                              configuration.
      --receive.hashrings-file-refresh-interval=5m
                              Refresh interval to re-read the hashring
                              configuration file. (used as a fallback)

>       --receive.local-endpoint=RECEIVE.LOCAL-ENDPOINT

Reviewer: This is not related to this PR but a bit related to your work with the router.  To me, this flag doesn't make any sense after the split. Is this flag actually used in the code?

---

> title: Receive-route

Reviewer 2: ```suggestion
title: receive-route
```

type: docs
menu: components
---

# Receive-route

The `thanos receive-route` command implements the [Prometheus Remote Write API](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write). It receives metrics from the target and forwards them to the right [receiver](./receive.md) in the hashring. It also buffers the incoming requests and handles changes in hashring configurations.

> We recommend this component to users who can only push into a Thanos due to air-gapped, or egress only environments. Please note the [various pros and cons of pushing metrics](https://docs.google.com/document/d/1H47v7WfyKkSLMrR8_iku6u9VB73WrVzBHb2SB6dL9_g/edit#heading=h.2v27snv0lsur).

Reviewer 3: small nit:
```suggestion
We recommend this component to users who can only push into a Thanos due to air-gapped or egress-only environments. Please note the [various pros and cons of pushing metrics](https://docs.google.com/document/d/1H47v7WfyKkSLMrR8_iku6u9VB73WrVzBHb2SB6dL9_g/edit#heading=h.2v27snv0lsur).
```

For further information on tuning Prometheus Remote Write [see remote write tuning document](https://prometheus.io/docs/practices/remote_write/).

# Example

```bash
thanos receive-route \
    --http-address 0.0.0.0:10909 \
    --receive.replication-factor 1 \
    --receive.hashrings-file ./data/hashring.json \
    --remote-write.address 0.0.0.0:10908 \
```

The example of `remote_write` Prometheus configuration:

```yaml
remote_write:

> - url: http://<thanos-receive-container-ip>:10908/api/v1/receive

Reviewer 3: We should point Prometheus at the router rather than directly at the receiver:
```suggestion
- url: http://<thanos-receive-route-container-ip>:10908/api/v1/receive
```

```

> where `<thanos-receive-containter-ip>` is an IP address reachable by Prometheus Server.

Reviewer 3: ```suggestion
where `<thanos-receive-route-containter-ip>` is an IP address reachable by Prometheus Server.
```

The example content of `hashring.json`:

```json
[
    {
        "endpoints": [
            "127.0.0.1:10907",
            "127.0.0.1:11907",
            "127.0.0.1:12907"
        ]
    }
]
```

> With such configuration, router listens for remote write on `<ip>10908/api/v1/receive` and will forward to the correct receiver in hashring

Reviewer 4: missing a color `:`

> With such configuration, router listens for remote write on `<ip>10908/api/v1/receive` and will forward to the correct receiver in hashring

Reviewer 3: Grammar nit: we should use the articles here and keep verbs in the same tense
```suggestion
With such configuration, the router will listen for remote write requests on `<ip>10908/api/v1/receive` and will forward to the correct receiver in the hashring
```

for tenancy and replication.

## Flags

> [embedmd]:# (flags/receive.txt $)

Reviewer 3: How was this generated? This looks like embedmd may try to template in the help text from the receiver vs the receive router

                              gRPC certificates. See
                              https://tools.ietf.org/html/rfc4366#section-3.1
      --receive.hashrings-file=<path>
                              Path to file that contains the hashring
                              configuration.
      --receive.hashrings-file-refresh-interval=5m
                              Refresh interval to re-read the hashring
                              configuration file. (used as a fallback)
      --receive.local-endpoint=RECEIVE.LOCAL-ENDPOINT
                              Endpoint of local receive node. Used to identify
                              the local node in the hashring configuration.
      --receive.tenant-header="THANOS-TENANT"
                              HTTP header to determine tenant for write
                              requests.
      --receive.default-tenant-id="default-tenant"
                              Default tenant ID to use when none is provided via
                              a header.
      --receive.replication-factor=1
                              How many times to replicate incoming write
                              requests.

```

> 

Reviewer 3: Ditto here with the newlines
